query/src/lib.rs

```diff
@@ -8,4 +8,206 @@
 // CONDITIONS OF ANY KIND, either express or implied. See the License for the
 // specific language governing permissions and limitations under the License.
 
-pub mod find;
+///! This module defines some core types that support find expressions: sources,
+///! variables, expressions, etc.
+///! These are produced as 'fuel' by the query parser, consumed by the query
+///! translator and executor.
+///!
+///! Many of these types are defined as simple structs that are little more than
+///! a richer type alias: a variable, for example, is really just a fancy kind
+///! of string.
+///!
+///! At some point in the future, we might consider reducing copying and memory
+///! usage by recasting all of these string-holding structs and enums in terms
+///! of string references, with those references being slices of some parsed
+///! input query string, and valid for the lifetime of that string.
+///!
+///! For now, for the sake of simplicity, all of these strings are heap-allocated.
+///!
+///! Furthermore, we might cut out some of the chaff here: each time a 'tagged'
+///! type is used within an enum, we have an opportunity to simplify and use the
+///! inner type directly in conjunction with matching on the enum. Before diving
+///! deeply into this it's worth recognizing that this loss of 'sovereignty' is
+///! a tradeoff against well-typed function signatures and other such boundaries.
+
+extern crate edn;
+extern crate num;
+extern crate ordered_float;
+
+use num::BigInt;
+use ordered_float::OrderedFloat;
+use edn::{Keyword, NamespacedKeyword, PlainSymbol};
+
+pub type SrcVarName = String;          // Do not include the required syntactic '$'.
+
+pub struct Variable(pub PlainSymbol);
+
+#[derive(Clone,Debug,Eq,PartialEq)]
+pub enum SrcVar {
+    DefaultSrc,
+    NamedSrc(SrcVarName),
+}
+
+/// These are the scalar values representable in EDN.
+#[derive(Clone,Debug,Eq,PartialEq)]
+pub enum NonIntegerConstant {
```

Comment:
It's not there, for a reason — an integer could be only an entid in some contexts (in the attribute place, for example) or either an entid or an integer in others. Those distinctions are modeled by the choice to use `Entid` or `EntidOrInteger` in conjunction with `NonIntegerConstant`.

There are other ways to slice this up, and I'll probably end up changing this somewhat when I flesh out the rest of the parser and start consuming these things.